MainProject/Company.cs

```diff
@@ -4,73 +4,102 @@ namespace MainProject
+
+        public Shop[] Shops
         {
+            get { return _shops; }
+            private set { _shops = value; }
         }
 
-        public Company(string name, int year)
+        public static int MinimumCarCount { get; private set; }
+
+        public Company(string name, int year, string vatNumber)
         {
-            this.name = name;
-            foundationYear = year;
+            Name = name;
+            FoundationYear = year;
+            _vatNumber = vatNumber;
         }
 
-        public Company(string name, int year, Shop shop)
+        public Company(string name, int year, Shop shop, string vatNumber)
         {
-            this.name = name;
-            foundationYear = year;
-            shops = new[] { shop };
+            Name = name;
+            FoundationYear = year;
+            Shops = new[] { shop };
+            _vatNumber = vatNumber;
         }
 
-        public Company(string name, int year, Shop[] shops)
+        public Company(string name, int year, Shop[] shops, string vatNumber)
         {
-            this.name = name;
-            foundationYear = year;
-            this.shops = shops;
+            Name = name;
+            FoundationYear = year;
+            Shops = shops;
+            _vatNumber = vatNumber;
         }
 
-        public string GetCompanyInfo() => $"Company {name} was established in {foundationYear}";
+        static Company()
+        {
+            MinimumCarCount = 2;
+        }
+
+        public virtual string GetCompanyInfo() => $"Company {Name} was established in {FoundationYear} and VAT number {_vatNumber}";
 
-        public void PrintCompanyName() => Console.WriteLine($"Company name is {name}");
+        public void PrintCompanyName() => Console.WriteLine($"Company name is {Name}");
 
-        public int GetShopCount() => shops.Length;
+        public int GetShopCount() => Shops.Length;
 
         public void AddShop(Shop shop)
         {
-            Array.Resize(ref shops, shops.Length + 1);
-            shops[shops.Length - 1] = shop;
+            if (_shops == null)
+            {
+                _shops = new[] { shop };
+            }
+            else
+            {
+                Array.Resize(ref _shops, _shops.Length + 1);
+                _shops[_shops.Length - 1] = shop;
+            }
         }
 
         public void AddShop(string shopName, int year)
         {
             Shop shop = new Shop(shopName, year);
-            Array.Resize(ref shops, shops.Length + 1);
-            shops[shops.Length - 1] = shop;
+            Array.Resize(ref _shops, _shops.Length + 1);
+            _shops[_shops.Length - 1] = shop;
         }
 
         public void PrintAllShops()
         {
-            foreach (Shop shop in shops)
+            Console.WriteLine($"Company {Name} shops: ");
+            foreach (Shop shop in _shops)
             {
                 shop.PrintShopName();
             }
+            Console.WriteLine();
```

Comment:
Leave empty line after brackets ({} - blocks of code - conditions, loops). Empty line before is optional

Reply:
Done